src/html-grammar.pegjs

```diff
@@ -548,6 +549,45 @@ text_node "Text Node"
   };
 }
 
+/* Inline php code */
+php "PHP Code"
+  = php_open com:(php_block) cc:(php_close)?
```

Comment:
You can't have `php_close` be an optional item (e.g. `?`) if you want to match the end tag _or_ the end of the file.